src/pages/ca-checkbook/forms/CheckbookForm.js

```diff
@@ -0,0 +1,174 @@
+import { Grid } from '@mui/material'
+import { useContext, useEffect } from 'react'
+import * as yup from 'yup'
+import FormShell from 'src/components/Shared/FormShell'
+import toast from 'react-hot-toast'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { useInvalidate } from 'src/hooks/resource'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import CustomTextField from 'src/components/Inputs/CustomTextField'
+import { useForm } from 'src/hooks/form'
+import { ControlContext } from 'src/providers/ControlContext'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { CashBankRepository } from 'src/repositories/CashBankRepository'
+import { ResourceLookup } from 'src/components/Shared/ResourceLookup'
+import CustomDatePicker from 'src/components/Inputs/CustomDatePicker'
+import CustomNumberField from 'src/components/Inputs/CustomNumberField'
+import { formatDateFromApi, formatDateToApi } from 'src/lib/date-helper'
+
+export default function CheckbookForm({ labels, maxAccess, recordId }) {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+
+  const invalidate = useInvalidate({
+    endpointId: CashBankRepository.CACheckbook.page
+  })
+
+  const { formik } = useForm({
+    initialValues: {
+      recordId: null,
+      bankAccountId: null,
+      size: '',
+      firstCheckNo: '',
+      lastCheckNo: '',
+      issueDate: null,
+      bankAccountRef: null,
+      bankAccountName: null
+    },
+    maxAccess,
+    enableReinitialize: true,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      bankAccountId: yup.string().required(),
+      size: yup.number().min(0).max(99).required(),
```

Comment:
when the size is greater than the limit the field is red with no message; i guess the user will not understand how to fix this